erpnext/controllers/status_updater.py

```diff
@@ -107,17 +26,32 @@ def update_prevdoc_status(self):
 		self.validate_qty()
 
 	def set_status(self, update=False, status=None, update_modified=True):
+		"""
+			{
+				Doctype: [
+					[status1, condition1],
+					[status2, condition2]
+				]
+			}
+		"""
 		if self.is_new():
 			if self.get('amended_from'):
 				self.status = 'Draft'
 			return
 
-		if self.doctype in status_map:
+		# try to make doctype key with status and condition as list of list. be careful before modifying this structure...
+		status_maps= frappe.get_all("Status Workflow", filters={"document_type":self.doctype}, fields=["*"])
+		records = {}
+		records = {item['document_type']:[] for item in status_maps}
+		for item in status_maps:
+			records[item['document_type']].append([item.status, item.condition])
```

Comment:
- Lets merge these two loops, apparently they are doing the same thing viz populating the records dict

Reply:
we need to maintain structure like { doctype: [
[status, condition]
]}
so modifying this not good option